cpp/src/replace/clamp.cu

```diff
@@ -0,0 +1,385 @@
+            } else {
+                memcpy(d_chars + d_offsets[idx], input_element.data(), input_element.size_bytes() );
+            }
+        };
+        thrust::for_each_n(exec->on(stream), thrust::make_counting_iterator<size_type>(0), strings_count, copy_transformer);
+
+        return make_strings_column(strings_count, std::move(offsets_column), std::move(chars_column),
+                input.null_count(), std::move(copy_bitmask(input.parent())), stream, mr);
+    }
+}
+
+template <typename T, typename InputIterator, typename ScalarZipIterator, typename Transformer>
+void apply_transform (InputIterator input_begin,
+        InputIterator input_end,
+        ScalarZipIterator scalar_zip_itr,
+        mutable_column_device_view output,
+        Transformer trans,
+        cudaStream_t stream)
+{
+    thrust::transform(rmm::exec_policy(stream)->on(stream),
+            input_begin,
+            input_end,
+            scalar_zip_itr,
+            output.begin<T>(),
+            trans);
+}
+
+template <typename T, typename ScalarIterator>
+std::enable_if_t<cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>>
+clamper(column_view const& input,
+        ScalarIterator const& lo_itr,
+        ScalarIterator const& lo_replace_itr,
+        ScalarIterator const& hi_itr,
+        ScalarIterator const& hi_replace_itr,
+        rmm::mr::device_memory_resource* mr,
+        cudaStream_t stream) {
+    auto output = detail::allocate_like(input, input.size(), mask_allocation_policy::NEVER, mr, stream);
+    // mask will not change
+    if (input.nullable()){
+        output->set_null_mask(copy_bitmask(input), input.null_count());
+    }
+
+    auto output_device_view  = cudf::mutable_column_device_view::create(output->mutable_view(), stream);
+    auto input_device_view  = cudf::column_device_view::create(input, stream);
+    auto scalar_zip_itr = thrust::make_zip_iterator(thrust::make_tuple(lo_itr, lo_replace_itr, hi_itr, hi_replace_itr));
+
+    auto trans = [] __device__ (auto element_validity_pair,
+                                auto scalar_tuple) {
+        if(element_validity_pair.second) {
+            auto lo_validity_pair = thrust::get<0>(scalar_tuple);
+            auto hi_validity_pair = thrust::get<2>(scalar_tuple);
+            if(lo_validity_pair.second and
+                (element_validity_pair.first < lo_validity_pair.first)) {
+                return thrust::get<1>(scalar_tuple).first;
+            } else if(hi_validity_pair.second and
+                (element_validity_pair.first > hi_validity_pair.first)) {
+                return thrust::get<3>(scalar_tuple).first;
+            }
+        }
+
+        return element_validity_pair.first;
+    };
+
+    if (input.has_nulls()) {
+        auto input_pair_iterator = make_pair_iterator<T, true>(*input_device_view);
+        apply_transform<T>(input_pair_iterator, input_pair_iterator+input.size(),
```

Comment:
Why do you need a wrapper around `thrust::transform` that does nothing but call `thrust::transform()`? Why not call it directly?

Reply:
Ahh, it was updated and missed to remove.